### ISSUE TITLE
ListView.DataSource doesn't use getSectionData method

### ISSUE THREAD

**Issue by tomprogers** [facebook/react-native#7067]
It appears to be the case that `ListView.DataSource` is ignoring the provided `getSectionData` method, with the result that `<ListView>`'s `renderSectionHeader` method is invoked with an undefined blob argument.

[Demo app on rnplay](https://rnplay.org/apps/kSwrYw)

Each section header is supposed to show the type of the `sectionData` argument passed to the `renderSectionHeader` method; at runtime, this is `undefined`. I would expect it to be an object:

* for the first header, this object: `{ "timestamp": "2016-04-18T05:00:00.000Z", "flavor": "tom rules" }`
* for the second header, this object `{ "timestamp": "2016-04-15T05:00:00.000Z", "flavor": "jake sux" }`

My expectations here are based on [this article](http://moduscreate.com/react-native-listview-with-section-headers/) which I used as a guide.

This bug prevents me from accessing the package of information I've placed in the blob for use when rendering section headers. With `sectionId` being the only datapoint available to me, my options are limited as far as data-driven header content.

Note: when running this code in a real simulator or on an actual phone, the log statements show that `getRowData` is invoked for every row, while `getSectionData` is never invoked. I'm not sure why these statements are not visible on rnplay.

Deets:
* `react-native v0.20.0`
* only tested on iOS so far
* developing on Mac

**Comment by grabbou** [comment]
The following works as expected -> https://rnplay.org/apps/5_NBzw. I have changed a bit the dataBlob you are passing to be an object and let ListViewDataSource calculate the rest for you. This is the cleanest impl. I am usually using in my apps. It passes a map with sectionID as keys.

However if you can't change the way your dataBlog looks like, here is your fixed example -> https://rnplay.org/apps/H68wEQ. However, sectionIds are in this case array indexes on a data blob which may complicate slightly the way you structure mockData.sections object. 

Basically the only thing I changed was `sectionIds` you pass to DataSource are now:
```js
"sectionIds": [0, 1],
```
because they are assumed to be keys on a dataBlob. Since you are passing an array of items by section, keys are in fact array keys!

Note that rowIdsBySection are also assumed to be keys on dataBlob and that works well at the moment, because you are passing a nested array:
```js
"rowIdsBySection": [
   [  // keys on section idx 0 ],
   [  // keys on section idx 1 ]
];
```

**Comment by grabbou** [comment]
It should be `getSectionHeaderData`, not `getSectionData`, see: https://github.com/facebook/react-native/blob/master/Libraries/CustomComponents/ListView/ListViewDataSource.js#L56
